packages/in_app_purchase/lib/src/in_app_purchase_connection/in_app_purchase_connection.dart

```diff
@@ -6,12 +6,19 @@ import 'dart:async';
 import 'dart:io';
 import 'app_store_connection.dart';
 import 'google_play_connection.dart';
+import 'product.dart';
 
 /// Basic generic API for making in app purchases across multiple platforms.
 abstract class InAppPurchaseConnection {
   /// Returns true if the payment platform is ready and available.
   Future<bool> isAvailable();
 
+  /// Query the product list.
+  ///
+  /// Returns a List of [Product] that each [Product] uniquely matches one valid identifier in [identifiers].
+  /// Any item in the `identifiers` that is not specified in App Store Connect or Google Play Console will be ignored.
+  Future<List<Product>> queryProductDetails(List<String> identifiers);
```

Comment:
nit: I would name this `query[product classname]`.

Reply:
SG